packages/ui/src/containers/ApplyByRoleContainer/ApplyByRoleContainer.test.tsx

```diff
@@ -1,12 +1,71 @@
-import { render } from "../../../utils/jest-apollo";
+import { getProject } from "@eden/package-mock";
+import userEvent from "@testing-library/user-event";
+import { RouterContext } from "next/dist/shared/lib/router-context";
+
+import { render, screen } from "../../../utils/jest-apollo";
+import { createMockRouter } from "../../../utils/test-utils/createMockRouter";
 import { ApplyByRoleContainer } from "./";
 
-describe("ApplyByRoleContainer", () => {
-  it("renders without throwing", () => {
-    const { container } = render(
-      <ApplyByRoleContainer onViewProject={(val) => console.log(val)} />
-    );
+jest.mock("react-confetti");
+
+test("User can get to the YOU DID IT modal ", async () => {
+  const user = userEvent.setup();
+
+  await render(
+    <RouterContext.Provider value={createMockRouter({})}>
+      <ApplyByRoleContainer
+        onViewProject={(val) => console.log(val)}
+        loadingProject={false}
+        project={getProject()}
+      />
+    </RouterContext.Provider>
+  );
+
+  //There are multiple roles on a page with multiple "more" buttons. Here I am clicking on the first one:
+  await user.click(screen.getAllByText("More")[0]);
+
+  /*
+   By clicking on the "More" button we open up the ApplyByRoleModal in which a user can fill out more information
+
+   --Bellow are the tests for this modal:
+ */
+  //User selects the timezone from the dropdown
+  await user.click(screen.getByRole("combobox"));
+  await user.click(screen.getByText("UTC-10"));
+  await expect(screen.getByDisplayValue("UTC-10")).toBeInTheDocument();
+
+  //User types in that they can work 69 hour/week
+  const hoursInput = screen.getByPlaceholderText("Hours");
+
+  await user.click(hoursInput);
+  await user.type(hoursInput, "69");
+  await expect(screen.getByDisplayValue("69")).toBeInTheDocument();
+
+  //User inputs their social handles
+
+  //Twitter
+  const twitterHandleInput = screen.getByPlaceholderText("Twitter Handle");
+
+  await user.click(twitterHandleInput);
+  await user.type(twitterHandleInput, "sbelka");
+  await expect(screen.getByDisplayValue("sbelka")).toBeInTheDocument();
+
+  //GitHub
+  const GitHubHandleInput = screen.getByPlaceholderText("Github Handle");
+
+  await user.click(GitHubHandleInput);
+  await user.type(GitHubHandleInput, "sbelka-1703");
+  await expect(screen.getByDisplayValue("sbelka-1703")).toBeInTheDocument();
+
+  //Telegram
+  const TelegramHandleInput = screen.getByPlaceholderText("Telegram Handle");
+
+  await user.click(TelegramHandleInput);
+  await user.type(TelegramHandleInput, "sbelka_1703");
```

Comment:
We shouldn't need to click before typing, which would also allow us to condense these variables too:

```suggestion
  await user.type(screen.getByPlaceholderText("Hours"), "69");
  await expect(screen.getByDisplayValue("69")).toBeInTheDocument();

  //User inputs their social handles

  //Twitter
  await user.type(screen.getByPlaceholderText("Twitter Handle", "sbelka");
  await expect(screen.getByDisplayValue("sbelka")).toBeInTheDocument();

  //GitHub
  await user.type(screen.getByPlaceholderText("Github Handle", "sbelka-1703");
  await expect(screen.getByDisplayValue("sbelka-1703")).toBeInTheDocument();

  //Telegram
  await user.type(screen.getByPlaceholderText("Telegram Handle", "sbelka_1703");
```

Reply:
When using `user.click()`, I was trying to simulate how the user would interact with a component. But it makes sense if we remove the `click` and leave the user action relevant to the element being tested within the modal. We are not losing any value from doing that. I will take this approach in my future testing.